user_guide/src/quickstart/intro.md

```diff
@@ -88,3 +88,137 @@ So the eager snippet above would become:
+### Add data to your `DataFrame`
+
+Add a column to the `DataFrame`'s collection. Using [`pl.lit()`](POLARS_PY_REF_GUIDE/api/polars.lit.html?#polars.lit) we can add a new column populated with a `Literal`.
+
+```python
+{{#include ../examples/dataframe/add_col.py:4:}}
+df.sample(3)
+```
+
+```text
+{{#include ../outputs/dataframe/df_column_added.txt}}
+```
+
+### Modify your `DataFrame`
+
+Replace data using conditional expressions and chaining methods.
+
+```python
+{{#include ../examples/dataframe/replace.py:4:}}
+df.tail()
+```
+
+```text
+{{#include ../outputs/dataframe/df_replaced.txt}}
+```
+
+### Update data types
+
+**Use the [`Categorical`](POLARS_PY_REF_GUIDE/api/polars.datatypes.Categorical.html?#polars.datatypes.Categorical) type and do some string ops**. With the `Categorical` data type we want to plan ahead here. If we want to perform common ops like `join`s on our cats, we'll need to create this data within the [`StringCache`](POLARS_PY_REF_GUIDE/api/polars.StringCache.html#polars.StringCache) context manager.
+
+```python
+with pl.StringCache():
+    # create your pl.Categorical here 👋
+    # ...
+```
+
+For the purposes of this walk-through we'll just use [`toggle_string_cache`](POLARS_PY_REF_GUIDE/api/polars.toggle_string_cache.html#polars.toggle_string_cache).
+
+```python
+# collect strings in same cache for Categorical comparisons
+pl.toggle_string_cache(True)
+
+{{#include ../examples/dataframe/datatypes.py:4:}}
+df.tail()
+```
+
+```text
+{{#include ../outputs/dataframe/df_new_datatypes.txt}}
+```
+
+As a `Polars` user, it's up to you how you formulate your workflow. `StringCache` context managers provide a clean approach to overhead management if you need it.
+
+👇 To **estimate** your `DataFrame`'s size [in bytes](POLARS_PY_REF_GUIDE/api/polars.DataFrame.estimated_size.html?#polars.DataFrame.estimated_size) and access its types.
+
+```python
+df.estimated_size(), df.dtypes
+```
+
+```text
+{{#include ../outputs/dataframe/df_info.txt}}
+```
+
+### Use a `join`
+
+Because we're using `toggle_string_cache` to create our `Categorical` data for each `DataFrame`, we can perform a `join` to bring in the IDs for each `color`.
+
+```python
+{{#include ../examples/dataframe/cat_join.py:5:}}
+color_ids
+```
+
+```text
+{{#include ../outputs/dataframe/color_ids.txt}}
+```
+
+### Use a `groupby`
+
+We have `datetime`s in this `DataFrame`. Perform `groupby` operations on your `DataFrame` at **embarrassingly parallel** speeds, another fundamental topic for a new `Polars` user.
+
+```python
+{{#include ../examples/dataframe/date_groupby.py:5:}}
+res.head()
+```
+
+```text
+{{#include ../outputs/dataframe/dates_groupby_res.txt}}
+```
+
+### Read from Excel files, write to CSV files
+
+Use the Python API's `xlsx2csv` [`read_excel`](POLARS_PY_REF_GUIDE/api/polars.read_excel.html#polars.read_excel) helper method for reading Excel files or read from CSVs directly with [`read_csv`](POLARS_PY_REF_GUIDE/api/polars.read_csv.html#polars.read_csv).
```

Comment:
I think we should not mention `xlx2csv` here. That's an implementation detail for an end user.